src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -115,64 +98,104 @@ public override async Task<dynamic> GetCertificatesInResourceGroupAsync(string s
+            if (string.IsNullOrWhiteSpace(resourceGroupName))
+                throw new ArgumentNullException(nameof(resourceGroupName));
+
             var sitesResult = await Get($"subscriptions/{subscriptionName}/resourceGroups/{resourceGroupName}/sites");
             return JsonConvert.DeserializeObject(sitesResult);
         }
 
         public override async Task<dynamic> GetSitesInServerFarmAsync(string subscriptionId, string serverFarmName)
         {
+            if (string.IsNullOrWhiteSpace(subscriptionId))
+                throw new ArgumentNullException(nameof(subscriptionId));
+            if (string.IsNullOrWhiteSpace(serverFarmName))
+                throw new ArgumentNullException(nameof(serverFarmName));
+
             var sitesResult = await Get($"subscriptionid/{subscriptionId}/serverfarm/{serverFarmName}/sites");
             return JsonConvert.DeserializeObject(sitesResult);
         }
 
         public override async Task<string> GetSiteWebSpaceNameAsync(string subscriptionId, string siteName)
         {
-            return await Get($"subscriptionid/{subscriptionId}/sitename/{siteName}/webspacename");
-        }
+            if (string.IsNullOrWhiteSpace(subscriptionId))
+                throw new ArgumentNullException(nameof(subscriptionId));
+            if (string.IsNullOrWhiteSpace(siteName))
+                throw new ArgumentNullException(nameof(siteName));
 
-        public override Task<string> GetStorageVolumeForSiteAsync(string stampName, string siteName)
-        {
-            throw new NotImplementedException();
+            return await Get($"subscriptionid/{subscriptionId}/sitename/{siteName}/webspacename");
         }
 
         public override async Task<string> GetWebspaceResourceGroupName(string subscriptionId, string webSpaceName)
         {
+            if (string.IsNullOrWhiteSpace(subscriptionId))
+                throw new ArgumentNullException(nameof(subscriptionId));
+            if (string.IsNullOrWhiteSpace(webSpaceName))
+                throw new ArgumentNullException(nameof(webSpaceName));
+
             return await Get($"subscriptionid/{subscriptionId}/webspacename/{webSpaceName}/resourcegroupname");
         }
 
         public override async Task<dynamic> GetSite(string siteName)
         {
+            if (string.IsNullOrWhiteSpace(siteName))
+                throw new ArgumentNullException(nameof(siteName));
+
             return await GetSiteInternal(null, siteName, null);
         }
 
         public override async Task<dynamic> GetSite(string stampName, string siteName)
         {
+            if (string.IsNullOrWhiteSpace(siteName))
+                throw new ArgumentNullException(nameof(siteName));
+
             return await GetSiteInternal(stampName, siteName, null);
         }
 
         public override async Task<dynamic> GetSite(string stampName, string siteName, string slotName)
         {
+            if (string.IsNullOrWhiteSpace(siteName))
```

Comment:
Since `GetSiteInternal` is already checking if sitename is empty, do we need to do this again here?

Reply:
Trying to fail as high in the call stack as possible - it makes the output clearer from a user perspective. 
So this is for a better debugging experience, the one in the internal call is a catch-all.